vendorconsent/tcf2/consent.go

```diff
@@ -10,10 +13,47 @@ func Parse(data []byte) (api.VendorConsents, error) {
 		return nil, err
 	}
 
+	var vendorConsents vendorConsentsResolver
+	var vendorLegitInts vendorConsentsResolver
+
+	var legIntStart uint
+	var pubRestrictsStart uint
 	// Bit 229 determines whether or not the consent string encodes Vendor data in a RangeSection or BitField.
 	if isSet(data, 229) {
-		return parseRangeSection(metadata)
+		vendorConsents, legIntStart, err = parseRangeSection(metadata, metadata.MaxVendorID(), 230)
```

Comment:
Please rename legIntStart for improved clarify. legInt doesn't immediately come across as standing for Legitimate Interest.